### ISSUE TITLE
What happens when Cluster autoscaler and Karpenter are used in a cluster at the same time

### ISSUE THREAD

**Issue by YangYiming919** [aws/karpenter#2543]
### What karpenter features are relevant?

Cluster autoscaler and Karpenter

### How should the docs be improved?

Who controls the cluster when Cluster autoscaler and Karpenter are deployed in the same cluster? 
Can I select them to control different node groups?

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by suraj2410** [comment]
Basically during our experience while migrating from cluster autoscaler to karpenter, we were running both together! 
However we were using different taints and tolerations for most of our nodegroups except some default ones
What happened to the default ones without any taints and tolerations with both running is the following:
- karpenter and autoscaler both will see a valid pod pending and start provisioning the node
- in most cases karpenter won and the pod got scheduled in the karpenter node
- the new node from autoscaler that came up either got some other pending pod around the same time or after some time just got deleted as autoscaler decided to remove it due to no workload

TL;DR Both will work together to spin up a node but usually karpenter node wins since it gets ready sooner than the autoscaler one

**Comment by marc-ostrow** [comment]
We've been running Karpenter 1.0.6 and CA 1.29.2 side-by-side for a couple of weeks now. We have one CA worker group that is not tainted, but all other worker groups are tainted, and a different key is used for Karpenter than for CA.  All Karpenter worker groups (Nodepool/EC2NodeClass) are tainted.

I'm noticing, during CI/CD, a very low occurrence of CA not being able to scale out a node after Karpenter has tried and failed to find a matching labeled NodePool (because the Pod was node selecting a CA worker group!)  Sometimes a single retry of running the Job is able to run, sometimes it may take several attempts, but it eventually works again. 

Quite frustrating.  In the process of moving our workloads over to Karpenter...
